xwiki-platform-core/xwiki-platform-oldcore/src/main/resources/xwiki.hbm.xml

```diff
@@ -156,8 +156,12 @@
             <key-property name="link" type="string">
                 <column name="XWL_LINK" index="XWLNK_LINK" length="766" />
             </key-property>
+            <key-property name="type" type="string">
+                <column name="XWL_TYPE" default="'document'" length="768" />
```

Comment:
Did you try on MySQL ? AFAIK the whole composite-id must be lower than the max KEY size (that's why "link" is 766, because it shares the key with docId)

Reply:
Does that mean that the `composite-id` size is already maxed out?

Reply:
The `type` column does not need to be that large, less that 20 chars should be enough but if we don't have 0 size available for the `type` that won't change anything.

Reply:
> One solution could be to replace the existing composite key (which is apparently not really great performance wide if I trust [stackoverflow.com/questions/8746207/1071-specified-key-was-too-long-max-key-length-is-1000-bytes](https://stackoverflow.com/questions/8746207/1071-specified-key-was-too-long-max-key-length-is-1000-bytes)), with a generated id

That's pretty much what we do in the document table (but with the ID being a hash, which is a bit dangerous, so better choose a safer id generation strategy).

Reply:
I've tested on maria, mysql and postgres with the table definition below and on my first tests it seems to be working well. Still need to check what's generated for the existing rows (which is not so important since we recreate all the entries during the migration). And to add a unique constraint on the former composite-id.
```
<class name="com.xpn.xwiki.doc.XWikiLink" table="xwikilinks">
  <id name="id" type="long" unsaved-value="undefined">
  <column name="XWL_ID" not-null="true"/>
  <generator class="assigned"/>
  </id>
  <property name="docId" column="XWL_DOC_ID" type="long"/>
  <property name="link" type="string" column="XWL_LINK" length="766"/>
  <property name="type" type="string" column="XWL_TYPE" length="100" />
  <property name="fullName" type="string" column="XWL_FULLNAME" length="768"/>
  <property name="attachmentName" type="string" column="XWL_ATTACHMENT_NAME" length="768"/>
</class>
```

Reply:
>   <property name="link" type="string" column="XWL_LINK" length="766"/>

Should be 768, the only reason why it was 766 was because of the composite-id.

Reply:
Does that mean that we can update the link field size to 768 for some db engines?

Reply:
Yes, my point is that if we get rid of the composite-id constraint, we should definitely migrate to 768.